src/pudl_archiver/archivers/nrelatb.py

```diff
@@ -23,18 +27,117 @@ class NrelAtbArchiver(AbstractDatasetArchiver):
+            og_filename = (
+                excel_url.split("/")[-1]
+                .lower()
+                .strip()
+                .replace(" ", "-")
+                .replace("_", "-")
+                .replace(f"{year}-", "")
+                .replace(f"{atb_type}-", "")
+                .replace("atb-", "")
+                .replace("%20", "-")
+                .replace("annual-technology-baseline-", "")
+            )
+            excel_filename = f"nrelatb-{year}-{atb_type}-{og_filename}"
+            excel_url = urljoin(year_excel_url, excel_url)
+            await self.download_add_to_archive_and_unlink(
+                excel_url, excel_filename, zip_path
+            )
+            data_paths_in_archive.add(excel_filename)
+
+        return ResourceInfo(
+            local_path=zip_path,
+            partitions={"year": year, "data_type": atb_type},
+            layout=ZipLayout(file_paths=data_paths_in_archive),
+        )
+
+    async def get_transportation_resources(
+        self, year: int, atb_type: Literal["transportation"]
+    ):
+        """Get the transportation files."""
+        zip_path = self.download_directory / f"nrelatb-{year}-{atb_type}.zip"
+        data_paths_in_archive = set()
+
+        async def _clean_and_download_wow(parquet_file, parquet_dir):
+            parquet_filename = (
+                parquet_file.split("/")[-1]
+                .lower()
+                .strip()
+                .replace(" ", "-")
+                .replace("_", "-")
+            )
+            filename = f"nrelatb-{year}-{atb_type}-{parquet_filename}"
+            url = urljoin(parquet_dir, parquet_file)
+            await self.download_add_to_archive_and_unlink(url, filename, zip_path)
+
+        year_parquet_url = f"https://data.openei.org/s3_viewer?bucket=oedi-data-lake&prefix=ATB%2F{atb_type}%2Fparquet%2F{year}%2F"
+        parquet_pattern = re.compile(
+            rf"^https://oedi-data-lake.s3.amazonaws.com/ATB/{atb_type}/parquet/{year}/(.*).parquet$"
+        )
+        dir_pattern = re.compile(r"%2F$")
+        for parquet_dir in await self.get_hyperlinks(year_parquet_url, dir_pattern):
+            parquet_dir = urljoin(year_parquet_url, parquet_dir)
+            for parquet_file in await self.get_hyperlinks(parquet_dir, parquet_pattern):
+                await _clean_and_download_wow(parquet_file, parquet_dir)
+                data_paths_in_archive.add(parquet_file)
+
+            if not data_paths_in_archive:
+                for parquet_dir_rlly in await self.get_hyperlinks(
+                    parquet_dir, dir_pattern
+                ):
+                    parquet_dir_rlly = urljoin(parquet_dir, parquet_dir_rlly)
+                    for parquet_file in await self.get_hyperlinks(
+                        parquet_dir_rlly, parquet_pattern
+                    ):
+                        await _clean_and_download_wow(parquet_file, parquet_dir_rlly)
+                        data_paths_in_archive.add(parquet_file)
```

Comment:
this is extremely chaotic... it'd be nice to not have this be a nested hellscape...

Reply:
converted to a recursive method